osu.Desktop/Updater/AppImageUpdateManager.cs

```diff
@@ -0,0 +1,564 @@
+                            // NOTE: throws an Exception if the debugged AppImage is not made executable
+                            process.Start();
+                        }
+                    };
+                    preparedToRestart = true;
+                }
+            });
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+        }
+        /// <inheritdoc cref="AppImageUpdateTool.IsInstalled"/>
+        public static bool IsInstalled
+        {
+            get
+            {
+                return AppImageUpdateTool.IsInstalled();
+            }
+        }
+
+        private class UpdateCompleteNotification : ProgressCompletionNotification
+        {
+            [Resolved]
+            private OsuGame game { get; set; }
+
+            public UpdateCompleteNotification(AppImageUpdateManager updateManager)
+            {
+                Text = @"Update ready to install. Click to restart!";
+
+                Activated = () =>
+                {
+                    updateManager.PrepareUpdateAsync()
+                                 .ContinueWith(_ => updateManager.Schedule(() => game?.AttemptExit()));
+                    return true;
+                };
+            }
+        }
+
+        private class UpdateProgressNotification : ProgressNotification
+        {
+            private readonly AppImageUpdateManager updateManager;
+
+            public UpdateProgressNotification(AppImageUpdateManager updateManager)
+            {
+                this.updateManager = updateManager;
+            }
+
+            protected override Notification CreateCompletionNotification()
+            {
+                return new UpdateCompleteNotification(updateManager);
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours)
+            {
+                IconContent.AddRange(new Drawable[]
+                {
+                    new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Colour = ColourInfo.GradientVertical(colours.YellowDark, colours.Yellow)
+                    },
+                    new SpriteIcon
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Icon = FontAwesome.Solid.Upload,
+                        Colour = Color4.White,
+                        Size = new Vector2(20),
+                    }
+                });
+            }
+
+            public override void Close()
+            {
+                // cancelling updates is not currently supported by the underlying updater.
+                // only allow dismissing for now.
+
+                switch (State)
+                {
+                    case ProgressNotificationState.Cancelled:
+                        base.Close();
+                        break;
+                }
+            }
+        }
```

Comment:
Probably want these shared between here and `SquirrelUpdateManager`, and move methods like `PrepareUpdateAsync` which are used here in the notification to the base `UpdateManager` class as a virtual and override in both managers.